falcon/testing/helpers.py

```diff
@@ -327,11 +331,17 @@ async def collect(self, event: Dict[str, Any]):
                 if not isinstance(value, bytes):
                     raise TypeError('ASGI header names must be byte strings')
 
+                # NOTE(vytas): Ported basic validation from wsgiref.validate.
+                if not self._HEADER_NAME_RE.match(name):
+                    raise ValueError('Bad header name: {!r}'.format(name))
+                if self._BAD_HEADER_VALUE_RE.search(value):
+                    raise ValueError('Bad header value: {!r}'.format(value))
+
                 name_decoded = name.decode()
```

Comment:
missed a decode here

Reply:
Actually not, but maybe I could add a `NOTE`? The regex above rules out any wonky characters.